### PR TITLE
Micro benchmark fixes

### PR DESCRIPTION
**Pull request by adamsitnik**
Fixes issues reported offline by @fanyang-mono , requires a BenchmarkDotNet update (https://github.com/dotnet/BenchmarkDotNet/pull/2231)

### CODE REVIEW

File: src/Core/tests/Benchmarks.Droid/MainInstrumentation.cs [hunk 1]
```diff
@@ -67,20 +53,4 @@ static bool Run()
 		}
 		return success;
 	}
-
-	// NOTE: the built-in ConsoleLogger throws PlatformNotSupportedException
-	class Logger : ILogger
```
Comment (jonathanpeppers) [inline src/Core/tests/Benchmarks.Droid/MainInstrumentation.cs:72]: Was there a fix in BDN for the logger? My memory is that some `System.Console` API threw a `PlatformNotSupported` exception on Android. I don't remember if I sent a PR for it.

If you ran this on Android and it worked, we're good. I think it threw immediately when benchmarks started.

**Review by jonathanpeppers — APPROVED** [review]
Ok, I see the logger fix, looks good:

https://github.com/dotnet/BenchmarkDotNet/pull/2231/files#diff-730015564449fcfc208e1ac9c8376f683cc147372799d4eb8f61131deeba1f24R17

File: NuGet.config [hunk 2]
```diff
@@ -22,6 +22,8 @@
     <!-- Added manually for dotnet/runtime 6.0.11 -->
     <add key="darc-pub-dotnet-emsdk-c3fc739" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/darc-pub-dotnet-emsdk-c3fc739c/nuget/v3/index.json" />
     <add key="darc-pub-dotnet-runtime-1e1f688" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/darc-pub-dotnet-runtime-1e1f688d/nuget/v3/index.json" />
+    <!-- used only for preview package validation -->
+    <add key="bdn-ci" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />
```
Comment (jonathanpeppers) [inline NuGet.config:26]: @mattleibow can we merge this, if it adds a nightly feed?

We could try using `$(RestoreAdditionalProjectSources)` in `Benchmarks.Droid.csproj` instead?

Comment (fanyang-mono) [reply NuGet.config:26]: I believe that it is for my local testing, isn't it? @adamsitnik

Comment (adamsitnik) [reply NuGet.config:26]: > I believe that it is for my local testing, isn't it? 

That is correct. Once you confirm that it works we are going to upload new version of BDN to internal Microsoft BDN feed

**Review by mattleibow — APPROVED** [review]

**Pull request merged**